### ISSUE TITLE
Script does not detect failure to build of libstdc++ on macOS High Sierra

### ISSUE THREAD

**Issue by EraYaN** [rdp/ffmpeg-windows-build-helpers#326]
```
/Users/erwin/ffmpeg-windows-build-helpers/sandbox/cross_compilers/pkgs/gcc/source/gcc-7.1.0/libstdc++-v3/include/precompiled/stdc++.h:117:10: fatal error: unordered_map: No such file or directory
 #include <unordered_map>
          ^~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [x86_64-w64-mingw32/bits/stdc++.h.gch/O2ggnu++0x.gch] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [all-recursive] Error 1
make[2]: *** [all] Error 2
make[1]: *** [all-target-libstdc++-v3] Error 2
make: *** [all] Error 2
error

See build.log for more details.
Done building (or already built) MinGW-w64 cross-compiler(s) successfully...

```
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=81797

Script does not properly detect this failure leading to the first other failure being with mfx_dispach about not finding the standard library.

Workaround is `--gcc-cpu-count=1` (and omitting the -j flag) but it is slow, better is to move to 7.4 or 8.1 . Maybe move to latest: https://github.com/Zeranoe/mingw-w64-build 

EDIT: The latest Zeranoe script does not build pthreads, breaking zvbi
Taking the prebuild from here https://sourceware.org/pub/pthreads-win32/pthreads-w32-2-9-1-release.zip

And copying the `Pre-built.2/dll/x64` and `Pre-built.2/lib/x64` contents to the `mingw/lib` and the `Pre-built.2/include` to the `mingw/include`
then
```
ln -s "./libpthreadGC2.a" "./libpthread.a"
ln -s "./pthreadGC2.dll" "./libpthread.dll"
```
Fixes that.

EDIT2:
libvpx and libpng also break:

Add `-fno-asynchronous-unwind-tables` to CFLAGS and CXXFLAGS

libvpx needs a patch to `vp8/common/threading.h`
before the use of `semaphore.h` include `sys/types.h` (line 76)

EDIT3: patch was already there but did not apply, did it manually and commented out the checkout of git otherwise it would overwrite. And then it will build. Not sure why the patch does not apply.

EDIT4: finding tesseract with pkg-config just fails, only fix is to disable tesseract. This also goes for a build on ubuntu (docker)

```
test_cc -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/tesseract -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/leptonica -L/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/lib
BEGIN /tmp/ffconf.gmAO74E1/test.c
    1	#include <tesseract/capi.h>
    2	#include <stdint.h>
    3	long check_TessBaseAPICreate(void) { return (long) TessBaseAPICreate; }
    4	int main(void) { int ret = 0;
    5	 ret |= ((intptr_t)check_TessBaseAPICreate) & 0xFFFF;
    6	return ret; }
END /tmp/ffconf.gmAO74E1/test.c
/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/bin/x86_64-w64-mingw32-gcc -D_ISOC99_SOURCE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -U__STRICT_ANSI__ -D__USE_MINGW_ANSI_STDIO=1 -D__printf__=__gnu_printf__ -D_WIN32_WINNT=0x0600 -D_POSIX_C_SOURCE=200112 -D_XOPEN_SOURCE=600 -DPIC -DCACA_STATIC -DMODPLUG_STATIC -DOPJ_STATIC -mtune=generic -O3 -DLIBTWOLAME_STATIC -DMODPLUG_STATIC -DCACA_STATIC -mtune=generic -O3 -std=c11 -fomit-frame-pointer -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -pthread -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -DLIBXML_STATIC -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/freetype2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libpng16 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libxml2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/fribidi -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/freetype2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libpng16 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -DLIBXML_STATIC -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libxml2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/freetype2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libpng16 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/bs2b -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -DLIBXML_STATIC -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/freetype2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libpng16 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libxml2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/freetype2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libpng16 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/freetype2 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/libpng16 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/fribidi -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/openjpeg-2.3 -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/opus -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/opus -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/tesseract -I/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/leptonica -L/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/lib -c -o /tmp/ffconf.gmAO74E1/test.o /tmp/ffconf.gmAO74E1/test.c
In file included from /tmp/ffconf.gmAO74E1/test.c:1:0:
/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/tesseract/capi.h:108:15: error: expected declaration specifiers or '...' before '*' token
 typedef bool (*TessCancelFunc)(void* cancel_this, int words);
               ^
/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/tesseract/capi.h:109:15: error: expected declaration specifiers or '...' before '*' token
 typedef bool (*TessProgressFunc)(ETEXT_DESC* ths, int left, int right, int top,
               ^
/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/tesseract/capi.h:404:72: error: unknown type name 'TessCancelFunc'
 TESS_API void TESS_CALL TessMonitorSetCancelFunc( ETEXT_DESC* monitor, TessCancelFunc cancelFunc );
                                                                        ^~~~~~~~~~~~~~
/ffmpeg-windows-build-helpers/sandbox/cross_compilers/mingw-w64-x86_64/x86_64-w64-mingw32/include/tesseract/capi.h:407:74: error: unknown type name 'TessProgressFunc'
 TESS_API void TESS_CALL TessMonitorSetProgressFunc( ETEXT_DESC* monitor, TessProgressFunc progressFunc );
                                                                          ^~~~~~~~~~~~~~~~
/tmp/ffconf.gmAO74E1/test.c: In function 'check_TessBaseAPICreate':
/tmp/ffconf.gmAO74E1/test.c:3:45: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
 long check_TessBaseAPICreate(void) { return (long) TessBaseAPICreate; }
                                             ^
ERROR: tesseract not found using pkg-config

```
Last part from ffbuild/config.log. Ran on an ubuntu 18.04 docker image with just the apt-gets the scripts tell you to. Might be an upstream issue. The same happend with my try on gcc 8.1 on macOS.

**Comment by rdp** [comment]
which version of gcc is this fixed in do you know?

**Comment by EraYaN** [comment]
6.5/7.4/8.1 according to the bug. 8.1 gives some other issues though. Especially with building it and the correct binutils version to use. The new Zeranoe script is very different and does not include pthreads.

The tesseract issue in indeed an upstream issue:

https://github.com/jb-alvarado/media-autobuild_suite/issues/848

They mention using this commit: https://github.com/tesseract-ocr/tesseract.git#commit=8d3f811

On a different note:
Also I can make PR for a fully contained Dockerfile including one click scripts. (using ubuntu 17.10 now for new enough nasm version so we can have multilib x264.) Although if tesseract is pinned to older commit 18.04 (new LTS) works as well.

**Comment by rdp** [comment]
OK pinned tesseract.

Can you publish docker images or what would be the utility there?

Dang gcc 7.4 isn't out yet? Or I'd bump to that I guess.

And thanks!

**Comment by EraYaN** [comment]
The docker is just a Dockerfile right now. The utility is that there is no impact on your environment and that you can control the build environment much better. (So solves the nasm problem)

Although docker filesystems are not the fastest, so gcc -pipe flag might help.

Also one might not need to build ones own toolchain even, but I haven't looked into it. There might be some on dockerhub.

**Comment by rdp** [comment]
A docker might be nice, I assume 18.04 is somehow preferable? :)

On Wednesday, June 6, 2018, Erwin de Haan <notifications@github.com> wrote:
> The docker is just a Dockerfile right now. The utility is that there is
no impact on your environment and that you can control the build
environment much better. (So solves the nasm problem)
>
> Although docker filesystems are not the fastest, so gcc -pipe flag might
help.
>
> Also one might not need to build ones own toolchain even, but I haven't
looked into it. There might be some on dockerhub.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci4.googleusercontent.com/proxy/7tWxNJKgj2DslegrcN4H2pCuj7XH5BMRM-xsaixiuJy_WNmOu6HKk8Ur2t12aKTbieR-g4xlyskjKUmkqvdKk-YI7IYfqkHVVFzN5UzQl-XNLgiUknH4yTScLYdawRtYUDgF6fapqbeTfpBCtHrVxsl43pxZtw=s0-d-e1-ft#https://github.com/notifications/beacon/AAAw0Howa1TE8jpoHfF1rRuZlTcfaYi8ks5t54JXgaJpZM4Uaf0P.gif>

**Comment by EraYaN** [comment]
18.04 is the latest LTS release, so it's good for many years.

**Comment by rdp** [comment]
OK merged it, thanks!

**Comment by rdp** [comment]
I was able to get a working build from HS the other day FWIW...is this still a roblem?

**Comment by rdp** [comment]
Oh it's a parallel issue? Attempted temp workaround...

**Comment by rdp** [comment]
7.4 has been released, I bumped it to 8.2 in the meantime, hopefully we're good now, thanks for the report.
